examples/generalized_linear_models/GLM-discrete-choice_models.myst.md

```diff
@@ -0,0 +1,824 @@
+
+```{code-cell} ipython3
+az.plot_dist_comparison(idata_m4, var_names=["beta_price"]);
+```
+
+We have explicitly set a negative prior on price and recovered a parameter specification more in line with the basic theory of rational choice. The effect of price should have a negative impact on utility. The flexibility of priors here is key for incorporating theoretical knowledge about the process involved in choice. Priors are important for building a better picture of the decision making process and we'd be foolish to ignore their value in this setting. 
+
+```{code-cell} ipython3
+fig, axs = plt.subplots(1, 2, figsize=(20, 10))
+ax = axs[0]
+counts = c_df.groupby("choice")["choiceId"].count()
+labels = c_df.groupby("choice")["choiceId"].count().index
+predicted_shares = idata_m4["posterior"]["p"].mean(dim=["chain", "draw", "obs"])
+ci_lb = idata_m4["posterior"]["p"].quantile(0.025, dim=["chain", "draw", "obs"])
+ci_ub = idata_m4["posterior"]["p"].quantile(0.975, dim=["chain", "draw", "obs"])
+ax.scatter(ci_lb, labels, color="k", s=2)
+ax.scatter(ci_ub, labels, color="k", s=2)
+ax.scatter(
+    counts / counts.sum(),
+    labels,
+    label="Observed Shares",
+    color="red",
+    s=100,
+)
+ax.scatter(
+    predicted_shares,
+    labels,
+    label="Predicted Mean",
+    color="green",
+    s=100,
+)
+ax.hlines(
+    labels,
+    ci_lb,
+    ci_ub,
+    label="Predicted 95% Interval",
+    color="black",
+)
+ax.legend()
+ax.set_title("Observed V Predicted Shares")
+az.plot_ppc(idata_m4, ax=axs[1])
+axs[1].set_title("Posterior Predictive Checks")
+ax.set_xlabel("Shares")
+ax.set_ylabel("Crackers");
+```
+
+We can now also recover the differences among individuals estimated by the model for particular cracker choices. More precisely we'll plot how the individual specific contribution to the intercept drives preferences among the cracker choices. 
+
+```{code-cell} ipython3
+idata_m4
+```
+
+```{code-cell} ipython3
+def get_brand(brand):
+    predicted = (
+        idata_m4["posterior"]["beta_individual"]
+        .sel(alts_intercepts=[brand])
+        .mean(dim=["chain", "draw"])
+    )
+    ci_lb = (
+        idata_m4["posterior"]["beta_individual"]
+        .sel(alts_intercepts=[brand])
+        .quantile(0.025, dim=["chain", "draw"])
+    )
+    ci_ub = (
+        idata_m4["posterior"]["beta_individual"]
+        .sel(alts_intercepts=[brand])
+        .quantile(0.975, dim=["chain", "draw"])
+    )
+
+    intervals_df = pd.DataFrame(
+        {
+            f"means_{brand}": predicted.values.flatten(),
+            f"lb_{brand}": ci_lb.values.flatten(),
+            f"ub_{brand}": ci_ub.values.flatten(),
+        }
+    )
+
+    return intervals_df
+
+
+interval_dfs = [get_brand(b) for b in ["nabisco", "keebler", "sunshine"]]
+interval_dfs = pd.concat(interval_dfs, axis=1)
+interval_dfs = interval_dfs.sort_values("means_nabisco")
+interval_dfs.head()
+```
```

Comment:
I am not opposed to DataFrames, here the data has already been reduced significantly, so it is not very cumbersome to convert to it. I always try to check the xarray alternative for two reasons.

First and foremost because if we identify tasks that are too difficult to do with xarray objects we can work on fixing that. It also helps me grok xarray (to the point now after some time I can confidently post xarray code I haven't run with some guarantees it will run or nearly run).

Second, because we all know numpy and pandas very well (or most of us quite well at least), so we definitely know how to do X with numpy/pandas, so converting our data to that is often faster/easier to do (even if more verbose) because we know after that we'll be able to do X, whereas if we have to fight with xarray and its docs it might end up taking twice as time. However, if the docs do this, I feel users get the impression that it is not worth it to fight with xarray objects/docs and have no easily accessible examples of how they are useful to them, which is incoherent with the deafult returned object being an InferenceData

Reply:
Makes total sense! 

Thanks again for your time on this one! I really appreciate it.